base/utfconvert.jl

```diff
@@ -0,0 +1,444 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+# Functions to convert to different UTF encodings
+
+# Quickly copy and set trailing \0
+@inline function fast_utf_copy(T::Type{UInt16}, len, dat)
+    @inbounds return UTF16String(setindex!(copy!(Vector{T}(len), dat), 0, len))
```

Comment:
Again, if it even saves a couple of bounds checks, having the `@inbounds` doesn't hurt anything, does?